packages/filesystem/src/browser/file-tree/file-tree.ts

```diff
@@ -96,6 +95,13 @@ export namespace FileStatNode {
     export function is(node: TreeNode | undefined): node is FileStatNode {
         return !!node && 'fileStat' in node;
     }
+
+    export function getUri(node: TreeNode | undefined): string {
```

Comment:
`FileStatNode` contains info about an underlying file, you don't need to parse it from ID.
```ts
    export function getUri(node: TreeNode | undefined): string | undefined {
        if (is(node)) {
            return node.fileStat.uri; // or node.uri if you need an instance of URI
        }
        return undefined;
    }
```